### PR TITLE
Organise vignettes in sections

### PR DESCRIPTION
**Pull request by averissimo**
# Pull Request

Related to:
- https://github.com/insightsengineering/coredev-tasks/issues/468

### Changes description

- Creates 3 sections:
  - Getting started
  - Delayed data loading
  - Other dataset connectors

### CODE REVIEW

**Comment by averissimo** [comment]
First jab at it, feel free to provide feedback and ideas

Question?
- Should the delayed data loading section include the Acronym `... (DDL)` or connector `... (dataset connector)`

image

File: vignettes/teal-data.Rmd [hunk 1]
```diff
@@ -48,9 +48,9 @@ The `teal.data` package provides many features to help specify your data:
...
-- [Pre-processing data](https://insightsengineering.github.io/teal.data/main/articles/preprocessing-data.html)
-- [Pre-processing `DDL` data ](https://insightsengineering.github.io/teal.data/main/articles/preprocessing-delayed-data.html)
-- [Checking for reproducibility](https://insightsengineering.github.io/teal.data/main/articles/reproducibility.html)
-- [Using python to create your data](https://insightsengineering.github.io/teal.data/main/articles/teal.data-with-python.html)
+- [Specifying relationships between your datasets](join-keys.html)
+- [Dynamically loading your data (`DDL`)](using-delayed-data-basic.html)
+- [Pre-processing data](preprocessing-data.html)
+- [Pre-processing `DDL` data ](preprocessing-delayed-data.html)
+- [Checking for reproducibility](reproducibility.html)
+- [Using python to create your data](teal.data-with-python.html)
```
[the remark this answers is not in the document]
Comment (m7pr) [reply vignettes/teal-data.Rmd:56]: @averissimo I think relative are better in here honestly

**Review by m7pr — APPROVED** [review]
Great work @averissimo

**Comment by m7pr** [comment]
Hey @averissimo can I have one more favour? Would you be able to delete dates from vignettes?
See this comment https://github.com/insightsengineering/teal.logger/pull/55#issuecomment-1699896071

**Comment by averissimo** [comment]
Yes @m7pr will do for teal.data!

:question::grey_question:  Do you need help doing that for any other package?

**Comment by averissimo** [comment]
@m7pr it seems like you already did it as part of the readme updates c0ab07d147cf52e06a13320f7e38995785ef0238

**Comment by m7pr** [comment]
Sorry for that!
Yes I'm doing that for everything that we are planning to release. I'm good

**Pull request merged**